packages/itwinui-react/src/core/Typography/Anchor/Anchor.tsx

```diff
@@ -4,15 +4,42 @@
  *--------------------------------------------------------------------------------------------*/
 import React from 'react';
 import cx from 'classnames';
-import { useTheme } from '../../utils';
+import {
+  PolymorphicComponentProps,
+  PolymorphicForwardRefComponent,
+  useTheme,
+} from '../../utils';
 import '@itwin/itwinui-css/css/anchor.css';
 
-export const Anchor = React.forwardRef<
-  HTMLAnchorElement,
-  React.ComponentPropsWithoutRef<'a'>
->(({ className, ...rest }, ref) => {
+type AnchorOwnProps = Record<never, never>;
+
+export type AnchorProps<T extends React.ElementType = 'a'> =
+  PolymorphicComponentProps<T, AnchorOwnProps>;
+
+type AnchorComponent = PolymorphicForwardRefComponent<'a', AnchorOwnProps>;
+
+/**
+ * A consistently styled anchor component.
+ *
+ * Supports polymorphic `as` prop for use with `Link` components from routers,
+ * or to render as a button.
+ *
+ * @example
+ * <Anchor href='/'>Home</Anchor>
+ * <Anchor href='/projects'>Projects</Anchor>
+ *
+ * @example
+ * <Anchor as={Link} to='/'>Home</Anchor>
+ * <Anchor as='button' onClick={() => {}}>click me</Anchor>
+ */
+export const Anchor = React.forwardRef((props, ref) => {
+  const { as: Element = 'a', className, ...rest } = props;
+
   useTheme();
-  return <a className={cx('iui-anchor', className)} ref={ref} {...rest} />;
-});
+
+  return (
+    <Element className={cx('iui-anchor', className)} ref={ref} {...rest} />
+  );
+}) as AnchorComponent;
```

Comment:
nit: instead of `as AnchorComponent`, you could add the type directly on L35 to Anchor (how it is in `Button`). 

```tsx
export const Anchor: AnchorComponent = React.forwardRef((props, ref) => {
```

Reply:
did a bit differently, see [`b07612f`](https://github.com/iTwin/iTwinUI/pull/1006/commits/b07612f3e8a8a7b87637a4c902c8f7672ed5ad32)